src/BlockEntities/HopperEntity.cpp

```diff
@@ -413,21 +416,89 @@ bool cHopperEntity::MoveItemsFromChest(cChunk & a_Chunk)
 		}
 
 		auto neighborAbsCoord = neighbor->RelativeToAbsolute(neighborRelCoord);
-		auto sideChest = static_cast<cChestEntity *>(neighbor->GetBlockEntity(neighborAbsCoord));
+		auto sideChest = static_cast<cChestEntity *>(
+			neighbor->GetBlockEntity(neighborAbsCoord));
 		if (sideChest == nullptr)
 		{
 			FLOGWARNING("{0}: A chest entity was not found where expected, at {1}", __FUNCTION__, neighborAbsCoord);
 		}
 		else
 		{
-			if (MoveItemsFromGrid(*sideChest))
+			second = true;
+		}
+
+		// If there is a second chest, determine the order to pull from
+		if (second)
+		{
+			// check x equivalency
+			if (neighborAbsCoord.x == chestPos.x)
 			{
-				return true;
+				// the "primary" chest is the one with the higher z or x value
+				if (neighborAbsCoord.z > chestPos.z)
+				{
+					// side is "primary" so pull from it first
+					if (MoveItemsFromGrid(*sideChest))
+					{
+						return true;
+					}
+					// main is secondary, pull from next
+					if (MoveItemsFromGrid(*mainChest))
+					{
+						return true;
+					}
+				}
+				else
+				{
+					if (MoveItemsFromGrid(*mainChest))
+					{
+						return true;
+					}
+					if (MoveItemsFromGrid(*sideChest))
+					{
+						return true;
+					}
+				}
+				return false;
+			}
+			// z equivalency
+			else
+			{
+				if (neighborAbsCoord.x > chestPos.x)
+				{
+					if (MoveItemsFromGrid(*sideChest))
+					{
+						return true;
+					}
+					if (MoveItemsFromGrid(*mainChest))
+					{
+						return true;
+					}
+				}
+				else
+				{
+					if (MoveItemsFromGrid(*mainChest))
+					{
+						return true;
+					}
+					if (MoveItemsFromGrid(*sideChest))
+					{
+						return true;
+					}
+				}
+				return false;
 			}
```

Comment:
The check for x versus z equivalency is pointless here and just makes the code longer - if you change the inner if condition to be `if ((neighborAbsCoord.x > chestPos.x) || (neighborAbsCoord.z > chestPos.z))` you could then remove the outer if and save half the code.